tests/acceptance/fetch_test.py

```diff
@@ -169,6 +169,15 @@ def test_fetch_headers(server_url):
     assert actual_headers.get(b'Foo') == [b'bar']
 
 
+def test_fetch_headers_tcp_nodelay_same_result(server_url):
+    headers = {'foo': ['bar']}
+    eventual_result = fido.fetch(server_url + ECHO_URL,
+                                 headers=headers,
+                                 tcp_nodelay=True)
```

Comment:
this and a bunch of other tests are duplicated code, have you had a look at @pytest.mark.parametrize()?